From 398de2efb0dfef95d2cf51b24009f914f96d0d6b Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Thu, 8 Sep 2016 16:57:52 +0700
Subject: [PATCH 1/5] gst145-omxbufferpool:Fixed memory corruption and bad access

Porting from a patch in Linux 3.10 environment made by Andrey Vostrikov <andrey.vostrikov@cogentembedded.com>

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 8 +++++---
 omx/gstomxvideodec.c   | 2 --
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index a361eac..4a2156f 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -523,9 +523,11 @@ gst_omx_buffer_pool_free_buffer (GstBufferPool * bpool, GstBuffer * buffer)

 #ifdef HAVE_MMNGRBUF
   vdbuf_data = (GstOMXVideoDecBufferData *) omx_buf->private_data;
-  for (i = 0; i < GST_VIDEO_MAX_PLANES; i++)
-    if (vdbuf_data->id_export[i] >= 0)
-      mmngr_export_end_in_user (vdbuf_data->id_export[i]);
+  if (vdbuf_data) {
+    for (i = 0; i < GST_VIDEO_MAX_PLANES; i++)
+      if (vdbuf_data->id_export[i] >= 0)
+        mmngr_export_end_in_user (vdbuf_data->id_export[i]);
+  }
 #endif

   g_slice_free (GstOMXVideoDecBufferData, omx_buf->private_data);
diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index bca11fd..1d61509 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -1333,8 +1333,6 @@ copy_frame (const GstVideoInfo * info, GstBuffer * outbuf)

 static void GstOMXBufCallbackfunc (struct GstOMXBufferCallback *release)
 {
-  gint i;
-
   if (!release)
     return;

--
1.9.1
